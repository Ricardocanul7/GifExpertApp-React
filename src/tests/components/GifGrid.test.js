import React from 'react';
import '@testing-library/jest-dom';
import { useFetchGifs } from '../../hooks/useFetchGifs';
const { shallow } = require("enzyme");
const { default: GifGrid } = require("../../components/GifGrid");
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
    const category = 'One Punch';

    test('debe de mostrarse correctamente', () => {
        useFetchGifs.mockReturnValue({
            data : [],
            loading : true
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de mostrar items cuando se cargan imagÃ©nes useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'http://localhost/cualquiercosa.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: 'DEF',
            url: 'http://localhost/cualquiercosa.jpg',
            title: 'Cualquier cosa'
        }];
        useFetchGifs.mockReturnValue({
            data : gifs,
            loading : false
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
})
