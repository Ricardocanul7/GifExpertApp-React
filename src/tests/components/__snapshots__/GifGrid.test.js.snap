// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imag√©nes useFetchGifs 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="http://localhost/cualquiercosa.jpg"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="Cualquier cosa"
      url="http://localhost/cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
